src/data/logbook.ts

```diff
@@ -6,3 +8,62 @@ export interface LogbookEntry {
   domain: string;
   context_user_id?: string;
 }
+
+const DATA_CACHE = {};
+const ALL_ENTITIES = "*";
+
+export const getLogbookData = (
+  hass: HomeAssistant,
+  startDate: string,
+  endDate: string,
+  entityId?: string
+) => {
+  if (!entityId) entityId = ALL_ENTITIES;
+
+  if (!DATA_CACHE[startDate]) DATA_CACHE[startDate] = [];
+  if (!DATA_CACHE[startDate][endDate]) DATA_CACHE[startDate][endDate] = [];
```

Comment:
`DATA_CACHE[startDate]` is an array

Reply:
This is all just a copy from `ha-logbook-data`... didn't really check the code 😱